### PR TITLE
Add in capability to repeatedly run the upscaler in a row

### PR DESCRIPTION
**Pull request by sean-bailey**
While the ESRGAN upscaler models compatible with stable-diffusion.cpp at this time are fixed in their upscaling rate, they do not discriminate on input image data. This means that, hypothetically, we could continuously run that upscaling model on the outputs of the previous upscaling run until we run out of memory.

I've tested this locally on my M1 MBP, and have successfully done txt2img and img2img generation resulting in image resolutions of 8192x8192. Bigger than that and I get an out-of-memory error, but hypothetically a large enough machine should be able to do this. 

This does *not* affect the image upscale multiplier inherent to the model, it simply runs the same fixed-multiplier model repeatedly (so for example, an `x4` model would take a 512x512 image upscaled to 2048x2048 on one pass, and then from 2048x2048 to 8192x8192 on a second pass, then 32k x 32k, then 128...)

The changes were designed to have minimal impact on any other code, so anything which currently leverages legacy versions should continue to function fine. If you want to use the new upscale repeating function, add

`--upscale-repeats <int >= 1>` 

to your command. Otherwise, by default, it will default to run the upscaler once.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,6 +189,7 @@ arguments:
                                      <= 0 represents unspecified, will be 1 for SD1.x, 2 for SD2.x
   --vae-tiling                       process vae in tiles to reduce memory usage
   --control-net-cpu                  keep controlnet in cpu (for low vram)
+  --canny                            apply canny preprocessor (edge detection)
```
Comment (phudtran) [inline README.md:192]: Is this supported in the PR?

Comment (sean-bailey) [reply README.md:192]: Canny wasn't something I modified in any way, I think it's how I modified the README itself that is showing the line addition for `--upscale-repeats`. Generally, functionality which was working before should still be working now as the upscaling process happens at the end, and this only repeats the upscale process.

Comment (leejet) [reply README.md:190]: This is introduced by https://github.com/leejet/stable-diffusion.cpp/pull/131

**Comment by leejet** [comment]
Great! Thank you for your contribution.

**Pull request merged**